src/code/z_demo.c

```diff
@@ -1,99 +1,1650 @@
+                globalCtx->envCtx.screenFillColor[2] = 0;
+                if (cmd->base == 3) {
+                    globalCtx->envCtx.screenFillColor[3] = (1.0f - temp_f0) * 255.0f;
+                } else {
+                    globalCtx->envCtx.screenFillColor[3] = 255.0f * temp_f0;
+                }
+                break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EAF20.s")
+            case 4:
+            case 8:
+                globalCtx->envCtx.screenFillColor[0] = 0;
+                globalCtx->envCtx.screenFillColor[1] = 0xFF;
+                globalCtx->envCtx.screenFillColor[2] = 0;
+                if (cmd->base == 4) {
+                    globalCtx->envCtx.screenFillColor[3] = (1.0f - temp_f0) * 255.0f;
+                } else {
+                    globalCtx->envCtx.screenFillColor[3] = 255.0f * temp_f0;
+                }
+                break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EAFE0.s")
+            case 9:
+                gSaveContext.unk_3F48 = 1;
+                break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EB1DC.s")
+            case 10:
+            case 11:
+                globalCtx->envCtx.screenFillColor[0] = 0;
+                globalCtx->envCtx.screenFillColor[1] = 0;
+                globalCtx->envCtx.screenFillColor[2] = 0;
+                if (cmd->base == 10) {
+                    globalCtx->envCtx.screenFillColor[3] = (1.0f - temp_f0) * 255.0f;
+                } else {
+                    globalCtx->envCtx.screenFillColor[3] = 255.0f * temp_f0;
+                }
+                break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EB364.s")
+            case 12:
+                globalCtx->envCtx.screenFillColor[0] = (160.0f * (1.0f - temp_f0));
+                globalCtx->envCtx.screenFillColor[1] = (160.0f * (1.0f - temp_f0));
+                globalCtx->envCtx.screenFillColor[2] = (u16)(160.0f * (1.0f - temp_f0));
+                globalCtx->envCtx.screenFillColor[3] = 255;
+                break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EB4B4.s")
+            case 13:
+                globalCtx->envCtx.screenFillColor[0] = (160.0f * temp_f0);
+                globalCtx->envCtx.screenFillColor[1] = (160.0f * temp_f0);
+                globalCtx->envCtx.screenFillColor[2] = (160.0f * temp_f0);
+                globalCtx->envCtx.screenFillColor[3] = 255;
+                break;
+        }
+    }
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/Cutscene_Command_TransitionFX.s")
+#endif
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EB6F8.s")
+// Command 0x5A: Camera
+s32 Cutscene_Command_Camera(GlobalContext* globalCtx, u8* cmd) {
+    s32 sp1C = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EBB68.s")
+    bcopy(cmd, &sp1C, 4);
+    cmd += 4;
+    if (func_8016A168() == 0) {
+        func_80161998(cmd, &sCutsceneCameraInfo);
+    }
+    return sp1C + 4;
+}
```

Comment:
What is this function actually doing? Do you know? Is this command used? It seems to be using the debug camera info, and not one of the 4 cameras or subcameras used for cutscenes in OoT?

Reply:
Yeah, it is used a lot. I think every cutscene data I have looked at has used it. Commenting it out disables any type camera movement.
My guess is this will be a full type of camera subcommands used by db_camera

Reply:
That's really curious. I thought DbCamera would have been unused in retail but it seems like it's really important for cutscenes. I wonder if it was reworked? I may have to bump up it's priority, now I'm curious